sw/device/tests/chip_power_idle_load.c

```diff
@@ -0,0 +1,351 @@
+  }
+
+  // Enable all PWM channels
+  CHECK_DIF_OK(dif_pwm_phase_cntr_set_enabled(&pwm, kDifToggleEnabled));
+
+  // PINMUX
+  for (int i = 0; i < PWM_PARAM_N_OUTPUTS; ++i) {
+    CHECK_DIF_OK(
+        dif_pinmux_output_select(&pinmux, kPinmuxMioOut[i], kPinmuxOutsel[i]));
+  }
+
+  LOG_INFO("SW_INFO: PWM active");
+
+  // OTP
+
+  static const dif_otp_ctrl_config_t otp_ctrl_config = {
+      .check_timeout = UINT32_MAX,
+      .integrity_period_mask = 0x1,
+      .consistency_period_mask = 0x1,
+  };
+
+  CHECK_DIF_OK(dif_otp_ctrl_configure(&otp_ctrl, otp_ctrl_config));
+
+  LOG_INFO("SW_INFO: OTP periodic checks active");
+
+  // AON Timer - activate in Watchdog mode & IRQ
+  static const uint64_t kTimeTillBark = 1000;
+
+  CHECK_DIF_OK(dif_aon_timer_watchdog_stop(&aon_timer));
+  CHECK_DIF_OK(dif_aon_timer_clear_wakeup_cause(&aon_timer));
+  expected_isr_handler = wdog_irq_handler;
+  nmi_fired = false;
+  nmi_state = (dif_rv_core_ibex_nmi_state_t){0};
+  CHECK_DIF_OK(
+      dif_rv_core_ibex_enable_nmi(&rv_core_ibex, kDifRvCoreIbexNmiSourceWdog));
+  uint32_t count_cycles =
+      aon_timer_testutils_get_aon_cycles_from_us(kTimeTillBark);
+  aon_timer_testutils_watchdog_config(&aon_timer, count_cycles, UINT32_MAX,
+                                      false);
+
+  LOG_INFO("SW_INFO: AON Timer active");
+
+  LOG_INFO("SW_INFO: all H/W is active");
+  // indicate S/W is idle
+  test_status_set(0xff20);
+  CHECK_DIF_OK(dif_gpio_write(&gpio, 2, 1));
+  wait_for_interrupt();
+
+  /* Note, if running on DV environment:
+  ** We wait here for the AON Timer to issue an NMI.
+  ** This will make the S/W exit with "true" and the Verilog test to finish with
+  ** "pass". Without this, the CPU will remain idle and the simulator will end
+  ** the test (due to simulation timeout) with "fail".
+  */
+
+  CHECK_DIF_OK(dif_gpio_write(&gpio, 2, 0));
+
+  // Prepare to exit
+
+  // We expect the watchdog bark interrupt to be pending on the peripheral side.
+  CHECK(irq_is_pending, "Expected watchdog bark interrupt to be pending");
+
+  // Check NMI previous state (before clearing):
+  CHECK(nmi_state.wdog_enabled && nmi_state.wdog_barked,
+        "WDOG NMI state check1 not expected! wdog_enable:%x, wdog_raised:%x",
+        nmi_state.wdog_enabled, nmi_state.wdog_barked);
+
+  CHECK_DIF_OK(dif_rv_core_ibex_get_nmi_state(
+      &rv_core_ibex, (dif_rv_core_ibex_nmi_state_t *)&nmi_state));
+  CHECK(nmi_state.wdog_enabled && !nmi_state.wdog_barked,
+        "WDOG NMI state check2 not expected! wdog_enable:%x wdog_raised:%x",
+        nmi_state.wdog_enabled, nmi_state.wdog_barked);
+
+  // Check that no external interrupts have occurred.
+  CHECK(ext_irq_fired == false, "Unexpected external interrupt triggered.");
+
+  pwrmgr_testutils_is_wakeup_reason(&pwrmgr, 0);
+
+  aon_timer_testutils_shutdown(&aon_timer);
+
+  test_status_set(kTestStatusInTest);
+
+  LOG_INFO("SW_INFO: exitting");
```

Comment:
Tip: the `LOG_INFO` already prefixes the logs with `INFO` (UVM_INFO in case of DV simulations). This log is probably not needed since it will be immediately followed by the "TEST PASSED" flag with the `return true` statement below.

Reply:
Done